// @supabase/ui/dist/esm/components/Accordion/Accordion2.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { createContext, useContext, useCallback } from 'react';
import AccordionStyles from './Accordion.module.css.js';
import IconChevronUp from '../Icon/icons/IconChevronUp/IconChevronUp2.js';
import Typography from '../Typography/Typography2.js';
import { Disclosure } from '../../node_modules/@headlessui/react/dist/components/disclosure/disclosure.esm.js';
import { Transition } from '../../node_modules/@headlessui/react/dist/components/transitions/transition.esm.js';

const AccordionContext = /*#__PURE__*/createContext({
  defaultActiveId: [],
  icon: jsx(IconChevronUp, {
    strokeWidth: 2
  }, void 0),
  iconPosition: 'right',
  onChange: undefined
});

function Accordion({
  children,
  className,
  defaultActiveId = [],
  icon = jsx(IconChevronUp, {
    strokeWidth: 2
  }, void 0),
  iconPosition = 'right',
  bordered,
  onChange
}) {
  let containerClasses = [AccordionStyles['sbui-accordion-container']];

  if (bordered) {
    containerClasses.push(AccordionStyles['sbui-accordion-container--bordered']);
  }

  if (className) {
    containerClasses.push(className);
  }

  const contextValue = {
    defaultActiveId,
    icon,
    iconPosition,
    onChange
  };
  return jsx(AccordionContext.Provider, Object.assign({
    value: contextValue
  }, {
    children: jsx("div", Object.assign({
      className: containerClasses.join(' ')
    }, {
      children: children
    }), void 0)
  }), void 0);
}

function Item({
  children,
  className,
  label,
  id
}) {
  const {
    defaultActiveId,
    icon,
    iconPosition,
    onChange
  } = useContext(AccordionContext);
  let panelClasses = [AccordionStyles['sbui-accordion-item__panel']];
  let buttonClasses = [AccordionStyles['sbui-accordion-item__button']];

  if (className) {
    buttonClasses.push(className);
  }

  const isDefaultActive = id ? defaultActiveId === null || defaultActiveId === void 0 ? void 0 : defaultActiveId.includes(id) : false;
  const handleOnChange = useCallback(open => () => {
    if (onChange) {
      onChange({
        id,
        label,
        open
      });
    }
  }, [onChange, id, label]);
  return jsx(Disclosure, Object.assign({
    defaultOpen: isDefaultActive
  }, {
    children: ({
      open
    }) => jsxs(Fragment, {
      children: [jsxs(Disclosure.Button, Object.assign({
        className: open ? `${buttonClasses.join(' ')} sbui-accordion-item__button--open` : buttonClasses.join(' ')
      }, {
        children: [iconPosition === 'left' && icon, jsx(Typography.Text, {
          children: label
        }, void 0), iconPosition === 'right' && icon]
      }), void 0), jsx(Transition, Object.assign({
        show: open,
        enter: AccordionStyles[`sbui-accordion-item__panel--enter`],
        enterFrom: AccordionStyles[`sbui-accordion-item__panel--enterFrom`],
        enterTo: AccordionStyles[`sbui-accordion-item__panel--enterTo`],
        leave: AccordionStyles[`sbui-accordion-item__panel--leave`],
        leaveFrom: AccordionStyles[`sbui-accordion-item__panel--leaveFrom`],
        leaveTo: AccordionStyles[`sbui-accordion-item__panel--leaveTo`],
        afterEnter: handleOnChange(open),
        afterLeave: handleOnChange(open)
      }, {
        children: jsx(Disclosure.Panel, Object.assign({
          className: panelClasses.join(' '),
          static: true
        }, {
          children: children
        }), void 0)
      }), void 0)]
    }, void 0)
  }), void 0);
}
Accordion.Item = Item;

export { Item, Accordion as default };
