// @radix-ui/react-popper/dist/index.module.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index_module$6 = require('../../react-arrow/dist/index.module.js');
var index_module$2 = require('../../react-primitive/dist/index.module.js');
var index_module$4 = require('../../react-use-size/dist/index.module.js');
var index_module$3 = require('../../react-use-rect/dist/index.module.js');
var index_module = require('../../react-context/dist/index.module.js');
var index_module$1 = require('../../react-compose-refs/dist/index.module.js');
var index_module$5 = require('../../popper/dist/index.module.js');
var React = require('react');
var _extends = require('../../../@babel/runtime/helpers/esm/extends.js');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () {
						return e[k];
					}
				});
			}
		});
	}
	n['default'] = e;
	return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

const[p,l]=index_module.createContext("Popper");const Popper=({children:e})=>{const[r,o]=React__namespace.useState(null);return React__namespace.createElement(p,{anchor:r,onAnchorChange:o},e)};const PopperAnchor=/*#__PURE__*/React__namespace.forwardRef(((e,o)=>{const{virtualRef:t,...n}=e,a=l("PopperAnchor"),p=React__namespace.useRef(null),f=index_module$1.useComposedRefs(o,p);return React__namespace.useEffect((()=>{a.onAnchorChange((null==t?void 0:t.current)||p.current);})),t?null:/*#__PURE__*/React__namespace.createElement(index_module$2.Primitive.div,_extends['default']({},n,{ref:f}))}));const[f,d]=index_module.createContext("PopperContent");const PopperContent=/*#__PURE__*/React__namespace.forwardRef(((e,n)=>{const{side:p="bottom",sideOffset:d,align:u="center",alignOffset:m,collisionTolerance:w,avoidCollisions:h=!0,...x}=e,v=l("PopperContent"),[A,g]=React__namespace.useState(),E=index_module$3.useRect(v.anchor),[y,C]=React__namespace.useState(null),P=index_module$4.useSize(y),[S,R]=React__namespace.useState(null),O=index_module$4.useSize(S),b=index_module$1.useComposedRefs(n,(e=>C(e))),z=function(){const[e,r]=React__namespace.useState(void 0);return React__namespace.useEffect((()=>{let e;function o(){r({width:window.innerWidth,height:window.innerHeight});}function t(){window.clearTimeout(e),e=window.setTimeout(o,100);}return o(),window.addEventListener("resize",t),()=>window.removeEventListener("resize",t)}),[]),e}(),T=z?DOMRect.fromRect({...z,x:0,y:0}):void 0,{popperStyles:k,arrowStyles:L,placedSide:B,placedAlign:D}=index_module$5.getPlacementData({anchorRect:E,popperSize:P,arrowSize:O,arrowOffset:A,side:p,sideOffset:d,align:u,alignOffset:m,shouldAvoidCollisions:h,collisionBoundariesRect:T,collisionTolerance:w}),H=void 0!==B;return React__namespace.createElement("div",{style:k,"data-radix-popper-content-wrapper":""},/*#__PURE__*/React__namespace.createElement(f,{arrowStyles:L,onArrowChange:R,onArrowOffsetChange:g},/*#__PURE__*/React__namespace.createElement(index_module$2.Primitive.div,_extends['default']({"data-side":B,"data-align":D},x,{style:{...x.style,animation:H?void 0:"none"},ref:b}))))}));const PopperArrow=/*#__PURE__*/React__namespace.forwardRef((function(r,o){const{offset:t,...n}=r,i=d("PopperArrow"),{onArrowOffsetChange:a}=i;return React__namespace.useEffect((()=>a(t)),[a,t]),/*#__PURE__*/React__namespace.createElement("span",{style:{...i.arrowStyles,pointerEvents:"none"}},/*#__PURE__*/React__namespace.createElement("span",{ref:i.onArrowChange,style:{display:"inline-block",verticalAlign:"top",pointerEvents:"auto"}},/*#__PURE__*/React__namespace.createElement(index_module$6.Root,_extends['default']({},n,{ref:o,style:{...n.style,display:"block"}}))))}));const Root=Popper;const Anchor=PopperAnchor;const Content=PopperContent;const Arrow=PopperArrow;

exports.Anchor = Anchor;
exports.Arrow = Arrow;
exports.Content = Content;
exports.Popper = Popper;
exports.PopperAnchor = PopperAnchor;
exports.PopperArrow = PopperArrow;
exports.PopperContent = PopperContent;
exports.Root = Root;
